src/Pos/Block/Logic/Internal.hs

```diff
@@ -111,8 +114,12 @@ type BlockApplyMode ssc m
 applyBlocksUnsafe
     :: forall ssc m . BlockApplyMode ssc m
     => OldestFirst NE (Blund ssc) -> Maybe PollModifier -> m ()
-applyBlocksUnsafe blunds0 pModifier =
-    reportingFatal version $
+applyBlocksUnsafe blunds0 pModifier = reportingFatal version $ do
+    when (length blunds /= length blunds0) $ assertionFailed $
```

Comment:
Yes, it's better than current approach and I will approve if you do it this way. But also can you rename `blunds0` to `blunds` and `b` to `blundsOtherEpoch` or something more descriptive?